### ISSUE TITLE
Impossible to get a token, errors on POST oauth2/token route

### ISSUE THREAD

**Issue by ceadreak** [mezzio/mezzio-authentication-oauth2#15]
### Bug Report

|    Q        |   A
|------------ | ------
| Version(s)  | 2.0.2

#### Summary

Impossible to get token with client_credentials & password grant

#### Current behavior

1/ On a Mezzio micro API, I try to get a new token with password grant. Got a **PostgreSQL** error : 
```
ERROR:  invalid input syntax for integer: "user_test"
STATEMENT:  INSERT INTO oauth_access_tokens (id, user_id, client_id, scopes, revoked, created_at, updated_at, expires_at) VALUES ($1, $2, $3, $4, $5, CURRENT_TIMESTAMP, CURRENT_TIMESTAMP, $6)
```

Should user_id be a VARCHAR ? In the oauth2.sql file, it's an integer

2/ I tried client_credentials grant too but got an error too : 
```
{
    "error": "invalid_client",
    "error_description": "Client authentication failed",
    "message": "Client authentication failed"
}
```

ClientEntity::$isConfidential is always false and in this case, the error above is emitted.

#### How to reproduce

Just try these two grants with Postman

#### Expected behavior

Get a token :)

**Comment by CedricDALAT** [comment]
Hi, i'm using 2.0.3 and the problem is still here, when you dont define the 'is_confidential' to true, the authentication is failing.
I found that pretty strange, you must say it's confidential to authenticate ?

**Comment by ceadreak** [comment]
Seems it was fixed in 2.0.3 : 

_adds support for checking if an OAuth2 client is marked as "confidential". To make use of this feature, you will need to add the field is_confidential to your oauth_clients table as a TINYINT: is_confidential tinyint(1) NOT NULL DEFAULT "0"_

**Comment by ZakariaAchaghour** [comment]
hello ,
I get the same issue I did not understand ??
